History/HistoryModel.cpp

```diff
@@ -350,28 +356,52 @@ void HistoryModel::scanInlineReferences()
             inlRef.mIsCurrent = false;
             inlRef.mIsStash = false;
         }
-        else if (ref == QLatin1String("refs/stash")) {
-            inlRef.mRefName = trUtf8( "<recent stash>" );
+        else if (ref == QLatin1Literal("refs/stash")) {
+            inlRef.mRefName = tr( "<recent stash>" );
             inlRef.mIsBranch = false;
-            inlRef.mIsCurrent = true;
+            inlRef.mIsCurrent = false;
             inlRef.mIsRemote = false;
             inlRef.mIsTag = false;
             inlRef.mIsStash = true;
         }
+        else if ( detached && ref == QLatin1Literal("< DETACHED >") ) {
+            inlRef.mRefName = tr("< DETACHED >");
+            inlRef.mIsBranch = false;
+            inlRef.mIsCurrent = false;
+            inlRef.mIsRemote = false;
+            inlRef.mIsTag = false;
+            inlRef.mIsStash = false;
+        }
         else {
             // qDebug() << "HistoryModel::scanInlineReferences => Unhandled ref:" << ref;
             continue;
         }
 
-        if (!refsById.contains(refs[ref])) {
-            refsById.insert(refs[ref], HistoryInlineRefs());
+        const Git::ObjectId &oid = refs[ref];
```

Comment:
`refs` is a `Git::ResolvedRefs`, a `Git::ResolvedRefs` is a `QHash`.

Now, `refs` can either be `const` (which is not the case here), then `QHash`'s `const` operator `[]` never returns a reference but always a copy!
Or it can be non-const (which is the case here), then `operator[]` won't return a `const` reference...

What will the compiler do with that? I'd assume that it _might_ chose to create a temporary object on the stack and point the `oid` reference to it. It might further chose that the temporary object isn't required anymore after this line...